### PR TITLE
fix: sorting new tag added in `TagsDialog`

### LINKED ISSUE #8764: [Bug] added tag in `TagsDialog` doesn't get sorted in the list correctly
###### Reproduction Steps

1. Have a list of tags (preffebly big)
2. Add a new tag "zzzzz" in `TagsDialog`.

###### Expected Result

All checked tags appearing first.

###### Actual Result

All checked tags appearing first except the new "zzzzz" tag

###### Research

This issue will be apparent once #8762 is fixed, this is not a regression bug since the legacy code had this same issue.

The issue is that the new tag is added to the list (as unchecked) then the list is sorted, after that the new tag is checked. 
This will result in the new tag belonging sorted with the unchecked tags since the sorting was done before checking the tag.

### PR DESCRIPTION
**Pull request by TarekkMA**
## Pull Request template
This PR is based on #8763

## Fixes
Fixes #8764

## Approach
sorting after the tag is checked

## How Has This Been Tested?

Emulator

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by TarekkMA** [comment]
Added a test  to check if the tag is sorted correctly when added

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.java [hunk 1]
```diff
@@ -279,12 +279,13 @@ protected void addTag(String tag) {
                     mNoTagsTextView.setVisibility(View.GONE);
                 }
                 mTags.add(tag);
+                mTags.check(tag);
                 mTagsArrayAdapter.sortData();
                 feedbackText = getString(R.string.tag_editor_add_feedback, tag, mPositiveText);
             } else {
+                mTags.check(tag);
```
Comment (TarekkMA) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.java:286]: Note to self: Extract this and do check+sort after if statement

Comment (TarekkMA) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.java:286]: Plus test if already existing tag is checked and sorted

Comment (TarekkMA) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/tags/TagsDialog.java]: fixed

**Review by mikehardy — APPROVED** [review]
Sorry for the delay! LGTM - thanks @TarekkMA

**Comment by mikehardy** [comment]
Hi there!

Apologies for the delay, and if you're already submitted something to OpenCollective for PRs merged in June 2021 PRs merged, you may feel free to ignore this and it will be processed shortly

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for June 2021 submissions

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

(I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged in May)

Thanks!

**Pull request merged**